pages/index.vue

```diff
@@ -147,15 +155,38 @@ export default defineComponent({
      * @todo 1. Send the request to update the task to the backend server.
      * @todo 2. Update the task in the dom.
      */
-    updateTask(_index, _id) {},
+    updateTask(_index, _id) {
+      let upTodo = this.msg[_index];
+      if(upTodo ==='' || upTodo === this.todos[_index].title){
+        this.editTask(_index);
+        return 
+      }
+      this.$toast.info('Updating Todo');
+      this.$axios({
+        headers: {Authorization: 'Token ' + this.$store.getters.token ,},
+        url: 'todo/' + (_id) + '/',
+        method: 'patch',
+        data: {
+          title: upTodo
+        }
+      })
+        .then(() =>{
+          this.$toast.success('Todo Updated!')
+          this.todos[_index].title = upTodo;
+          this.editTask(_index);
+        })
+    },
     /**
      * toggle visibility of input and buttons for a single todo
      * @argument {number} index - index of element to toggle
      * @todo add in bindings in dom so that 'hideme' class is dynamic or use conditional rendering
      * @hint read about class bindings in vue
      */
     editTask(index) {
-      this.todos[index].editing = !this.todos[index].editing
+      let a = this.todos[index];
+      this.msg[index] = a.title;
+      a.editing = !a.editing
+      this.todos.splice(index,1,a);
```

Comment:
this wasn't purpose of this function as defined in above comment. Also instead of a separate array you could just use just an addition variable which will store the string